chain/block.go

```diff
@@ -284,12 +312,32 @@ func (b *StatelessBlock) initializeBuilt(
 	b.results = results
 	b.feeManager = feeManager
 	b.txsSet = set.NewSet[ids.ID](len(b.Txs))
+
+	// block hash generation
+	var items2prove []merkletree.Content
+	// state root
+	items2prove = append(items2prove, TrieItem{x: b.StateRoot[:]})
+	// transactions
 	for _, tx := range b.Txs {
+		items2prove = append(items2prove, TrieItem{x: tx.Bytes()})
 		b.txsSet.Add(tx.ID())
 		if tx.WarpMessage != nil {
 			b.containsWarp = true
 		}
 	}
+	// transaction results
+	for _, result := range b.results {
+		items2prove = append(items2prove, TrieItem{x: result.Output})
+	}
+
+	trie, err := merkletree.NewTree(items2prove)
```

Comment:
We should use a stateless instance of the `MerkleDB` to build this trie, rather than this external package.

Reply:
Could a `MerkleDB` backed by a `memdb` necessarily be a stateless `MerkleDB`?

Reply:
Yeah, I'm looping in @dboehm-avalabs to assist here 👍

Reply:
Looks like it has already been updated, but let me know if you have any additional questions.